### PR TITLE
[FIX] Prevent torch 2.0.0 installation

### PR DESCRIPTION
**Pull request by JihwanEom**
### Summary
- Add `torchvision` & `torchtext` in `requirements/anomaly.txt` with fixed version (0.14.1) to prevent torch 2.0.0 installation

### How to test
With cuda 11.7 environment:
```
pip install torch==1.13.1 torchvision==0.14.1 --extra-index-url https://download.pytorch.org/whl/cu117
pip install -e .[full]
```

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: requirements/anomaly.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 # Anomaly Requirements.
+torchvision==0.14.1
```
Comment (goodsong81) [inline requirements/anomaly.txt]: Do we need this, too?

Comment (JihwanEom) [reply requirements/anomaly.txt]: Yes, without this, torchvision==0.15.1 and torch==2.0.0 are installed.

Comment (ashwinvaidya17) [reply requirements/anomaly.txt]: I am a bit hesitant about fixed requirements. Can't we do something like `torchvision<0.15.1`? Also, does `torchtext` need to be specified?

Comment (djdameln) [reply requirements/anomaly.txt]: What if we fix the torch version instead and let the installer sort out the torchvision and torchtext versions?

Comment (goodsong81) [reply requirements/anomaly.txt]: > What if we fix the torch version instead and let the installer sort out the torchvision and torchtext versions?

Currently, we regard the torch as pre-requisite. Users might want to use torch1.12.1. I would prefer @ashwinvaidya17's approach.

Comment (JihwanEom) [reply requirements/anomaly.txt]: umm.. I'm also not exactly sure about these changes. This PR is hotfix. (Absolutely opened for good ideas!)
If we can control the acceptable range of version in anomaly side, it looks best to me. https://github.com/openvinotoolkit/anomalib/blob/v0.4.0rc2/requirements/base.txt#L15-L16

Comment (goodsong81) [reply requirements/anomaly.txt]: > I am a bit hesitant about fixed requirements. Can't we do something like `torchvision<0.15.1`? Also, does `torchtext` need to be specified?

torchtext>=0.15.1 would install torchdata that would force to install torch==2.0. Looks like mistake.

Comment (goodsong81) [reply requirements/anomaly.txt]: image

File: requirements/anomaly.txt [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
+torchtext==0.14.1
```
Comment (samet-akcay) [inline requirements/anomaly.txt]: well, now that you just modified this file, maybe you could also replace `anomalib==0.4.0rc2` with `anomalib==0.4.0` too? 😁

**Comment by JihwanEom** [comment]
CI failed by codecov coverage upload issue, but passed all of unit tests. (known issue)

File: requirements/anomaly.txt (hunk 2 above)
Comment (JihwanEom) [reply requirements/anomaly.txt]: Reflected! :)

**Review by goodsong81 — APPROVED** [review]
Nice catch! Thank you for the quick fix.

**Pull request merged**